### ISSUE TITLE
Using IP Address Whitelisting With Heroku

### ISSUE THREAD

**Issue by hugeprestigellc** [jaggedsoft/node-binance-api#265]
I'm writing an app using this package and Heroku. In order to use IP address whitelisting for added security, I'm using a Heroku add-on that allows me to send traffic through a static IP.

Looking through the docs here, I thought I would be able to set the following environment variables to whatever my static IP was:
https_proxy
socks_proxy

But, for whatever reason, setting those variables is causing a lot of issues with other parts of my app. My guess is some other packages also uses those 2 variables and it's causing some issues.

I was however able to get IP Address whitelisting to work by modifying your code and adding my static URL to the request object like so:

const reqObj = (url, data = {}, method = 'GET', key) => ({
      url: url,
      proxy: process.env.QUOTAGUARDSTATIC_URL, // The line I added
      qs: data,
      method: method,
      timeout: options.recvWindow,
      headers: {
          'User-Agent': userAgent,
          'Content-type': contentType,
          'X-MBX-APIKEY': key || ''
      }
    })

This works, but obviously copying your code, modifying it and pushing it to my Heroku repo is not optimal.

Are there any plans to add a proxy URL option when using your package?

Thanks!

**Comment by jaggedsoft** [comment]
Yes this system could use a bit of a rework, I'll take a look at it. Thank you

**Comment by jaggedsoft** [comment]
I am currently in talks with Binance regarding a solution.
One fix I can think of that might make it easier for you is a 'heroku' option that applies your fix, that way you won't have to hack it

**Comment by dancherb** [comment]
Great, thanks - right now I've switched https_proxy and socks_proxy to https_proxy_binance and socks_proxy_binance in the module's source code and added that variable in .env as the QuotaGuard Static URL

**Comment by hugeprestigellc** [comment]
Hey @jaggedsoft, OP here, just wanted to chime in that I like the sound of that 'Heroku' option you mentioned.

I've implemented something similar to what @dancherb is currently but obviously this is less than ideal long term.

Thanks for all the great work with this by the way!

**Comment by amigoni** [comment]
I'm having an issue getting the Socket to work on Heroku. I'm trying to go through the socks_proxy and setting that variable in my env. But while I use Fixie in Heroku to get a static IP. The port keeps changing and I can't figure out how to add it dynamically to the socks_proxy env variable. Any tips you guys have? Thanks so much.
